src/Dropdown.php

```diff
@@ -4,359 +4,657 @@
+     * @return self A new instance with the specified attributes added.
+     */
+    public function addToggleAttributes(array $values): self
     {
-        $options = $this->prepareOptions();
+        $new = clone $this;
+        $new->toggleAttributes = array_merge($this->toggleAttributes, $values);
 
-        if (is_array($this->items)) {
-            return Html::ul()
-                ->attributes($options);
-        }
+        return $new;
+    }
 
-        $tagName = ArrayHelper::remove($options, 'tag', 'div');
+    /**
+     * Sets the HTML attributes for the dropdown component.
+     *
+     * @param array $values Attribute values indexed by attribute names.
+     *
+     * @return self A new instance with the specified attributes.
+     *
+     * @see {\Yiisoft\Html\Html::renderTagAttributes()} for details on how attributes are being rendered.
+     */
+    public function attributes(array $values): self
+    {
+        $new = clone $this;
+        $new->attributes = $values;
 
-        return Html::tag($tagName, '', $options)
-            ->encode($this->encodeTags);
+        return $new;
     }
 
-    public function begin(): ?string
+    /**
+     * Whether to automatically close the dropdown when any of its elements is clicked.
+     *
+     * @param bool $value If set to `true`, the dropdown will be closed when any of its elements is clicked. If set to
+     * `false`, should be closed manually.
+     *
+     * @return self A new instance with the specified auto-close setting.
+     */
+    public function autoCloseOnClick(bool $value = true): self
     {
-        parent::begin();
+        return $this->addToggleAttributes(['data-bs-auto-close' => $value === true ? 'true' : 'false']);
+    }
 
-        $options = $this->prepareOptions();
-        $this->tagName = is_array($this->items) ? 'ul' : ArrayHelper::remove($options, 'tag', 'div');
+    /**
+     * Whether to automatically close the dropdown when the user clicks inside of the dropdown.
+     *
+     * @param bool $value If set to `true`, the dropdown will be closed when the user clicks inside of the dropdown. If
+     * set to `false`, no attribute will be set.
+     *
+     * @return self A new instance with the specified auto-close setting.
+     */
+    public function autoCloseOnClickInside(bool $value = true): self
+    {
+        return $this->addToggleAttributes(['data-bs-auto-close' => $value === true ? 'inside' : null]);
+    }
 
-        return Html::openTag($this->tagName, $options);
+    /**
+     * Whether to automatically close the dropdown when the user clicks outside of the dropdown or the dropdown toggle
+     * button.
+     *
+     * @param bool $value If set to `true`, the dropdown will be closed when the user clicks outside of the dropdown. If
+     * set to `false`, no attribute will be set.
+     *
+     * @return self A new instance with the specified auto-close setting.
+     */
+    public function autoCloseOnClickOutside(bool $value = true): self
```

Comment:
I think that creating many enumerations can be confusing, in this particular case the methods are clear and expeditious.

Reply:
It's not clear. By fact for one property (`autoClose`) we have two methods, it's confusing.

Reply:
For me it's clear, each method explains and sets the correct value of the property, and has clear IDE support.

Reply:
This is not consistent with Bootstrap. Bootstrap has one property and four values. We also should to create one method to change this property.

Reply:
And because it is inconsistent, you are changing the value of the attribute, as the case may be, i don't see what the problem is.

Reply:
I think, it's problem. Must be one method only.

Reply:
It is a single attribute with 4 behaviors, each one is explained clearly and explicitly in each method, where the name of the method alone explains its behavior, unlike having a single method with 4 behaviors, for me that is more confusing.

Reply:
We couldn't use both method at the same time, so we need create one method instead of two.

Reply:
I'll actually change it just so i don't get stuck, if you use two chained methods the last value will remain, so it's not an error, there will always be only one behavior, and as for dynamic generation it's simpler and clearer to call a method than to add an enumeration.

Reply:
I'll actually change it just so I don't get stuck, but if you use two methods in a row you'll get the last behavior, so you'll only have one behavior, I designed it this way, so that dynamic generation is simpler using a method, than passing an enum to a method.

Reply:
done.